bookkeeper-server/src/main/java/org/apache/bookkeeper/server/service/AutoRecoveryService.java

```diff
@@ -56,8 +59,14 @@ public AutoRecoveryMain getAutoRecoveryServer() {
     protected void doStart() {
         try {
             this.main.start();
-        } catch (UnavailableException e) {
-            throw new RuntimeException("Can't not start '" + NAME + "' component.", e);
+        } catch (Throwable exc) {
+            LOG.error("Got unexpected exception while starting AutoRecoveryMain", exc);
```

Comment:
Lets move this to AbstractLifecycle#start() method, since exceptions can happen in any component start.

Reply:
changing that